compiler/src/iree/compiler/Codegen/SPIRV/test/pipeline_matvec.mlir

```diff
@@ -100,3 +108,121 @@ hal.executable @i4_dequant_matvec_f16 {
+    }
+    builtin.module {
+      func.func @i4_dequant_matvec_f16_subgroup_64() {
+        %cst = arith.constant 0.000000e+00 : f16 
+        %0 = hal.interface.constant.load[0] : i32 
+        %1 = hal.interface.constant.load[1] : i32 
+        %2 = hal.interface.constant.load[2] : i32 
+        %3 = hal.interface.constant.load[3] : i32 
+        %4 = hal.interface.constant.load[4] : i32 
+        %5 = arith.index_castui %0 : i32 to index 
+        %6 = arith.index_castui %1 : i32 to index 
+        %7 = arith.index_castui %2 : i32 to index 
+        %8 = arith.index_castui %3 : i32 to index 
+        %9 = arith.index_castui %4 : i32 to index 
+        %10 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%5) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<4096x86x128xi4>>
+        %11 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%6) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<4096x86xf16>>
+        %12 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%7) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<4096x86xf16>>
+        %13 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) alignment(64) offset(%8) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<86x128xf16>>
+        %14 = hal.interface.binding.subspan set(0) binding(2) type(storage_buffer) alignment(64) offset(%9) : !flow.dispatch.tensor<writeonly:tensor<4096xf16>>
+        %15 = flow.dispatch.tensor.load %10, offsets = [0, 0, 0], sizes = [4096, 86, 128], strides = [1, 1, 1] : !flow.dispatch.tensor<readonly:tensor<4096x86x128xi4>> -> tensor<4096x86x128xi4>
+        %16 = flow.dispatch.tensor.load %11, offsets = [0, 0], sizes = [4096, 86], strides = [1, 1] : !flow.dispatch.tensor<readonly:tensor<4096x86xf16>> -> tensor<4096x86xf16>
+        %17 = flow.dispatch.tensor.load %12, offsets = [0, 0], sizes = [4096, 86], strides = [1, 1] : !flow.dispatch.tensor<readonly:tensor<4096x86xf16>> -> tensor<4096x86xf16>
+        %18 = flow.dispatch.tensor.load %13, offsets = [0, 0], sizes = [86, 128], strides = [1, 1] : !flow.dispatch.tensor<readonly:tensor<86x128xf16>> -> tensor<86x128xf16>
+        %19 = tensor.empty() : tensor<4096xf16>
+        %20 = tensor.empty() : tensor<4096x86x128xf16>
+        %21 = linalg.fill ins(%cst : f16) outs(%19 : tensor<4096xf16>) -> tensor<4096xf16>
+        %22 = linalg.generic {indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d1, d2)>, affine_map<(d0, d1, d2) -> (d0, d1)>, affine_map<(d0, d1, d2) -> (d0, d1)>, affine_map<(d0, d1, d2) -> (d0, d1, d2)>], iterator_types = ["parallel", "parallel", "parallel"]} ins(%15, %16, %17 : tensor<4096x86x128xi4>, tensor<4096x86xf16>, tensor<4096x86xf16>) outs(%20 : tensor<4096x86x128xf16>) {
+        ^bb0(%in: i4, %in_0: f16, %in_1: f16, %out: f16): 
+          %24 = arith.extui %in : i4 to i32
```

Comment:
The patterns responsible for those rewrites are already tested upstream and it would just change the shift instruction used. I can add the test if it's still wanted though.

Reply:
OK, that's reasonable